pep-0458.txt

```diff
@@ -730,99 +735,97 @@ efficiently transfer consistent snapshots from PyPI.
-inconsistently.
-
-
-Snapshot Process
-----------------
-
-The snapshot process is fairly simple and SHOULD be automated.  The snapshot
-process SHOULD use the latest working set of the *targets* and all
-delegated targets roles (i.e. *bins* and *bin-n* roles) metadata.  Upon an update, the
-snapshot process will sign for this
-latest working set.  (Recall that project transaction processes continuously
-inform the snapshot process about the latest delegated metadata in a
-concurrency-safe manner.  The snapshot process will actually sign for a copy of
-the latest working set while the latest working set in memory will be updated
-with information that is continuously communicated by the project transaction
-processes.)  The snapshot process MUST generate and sign new *timestamp*
-metadata that will vouch for the metadata (*root*, *targets*, and delegated
-roles) generated in the previous step.  Finally, the snapshot process MUST make
-the new *timestamp* and *snapshot* metadata representing
-the latest snapshot available to clients.
-
-At this stage, its important to acknowledge a few implementation notes.  So far, we have seen only that
-new metadata and targets are added, but not that old metadata and targets are
-removed.  Practical constraints are such that eventually PyPI will run out of
-disk space to produce a new consistent snapshot.  In that case, PyPI MAY then
-use something like a "mark-and-sweep" algorithm to delete sufficiently old
-consistent snapshots. In order to preserve the latest consistent snapshot, PyPI
-should walk objects beginning from the root (*timestamp*) of the latest
-consistent snapshot, mark all visited objects, and delete all unmarked objects.
-The last few consistent snapshots may be preserved in a similar fashion.
-Deleting a consistent snapshot will cause clients to see nothing except HTTP
-404 responses to any request for a file within that consistent snapshot.
-Clients SHOULD then retry their requests (as before) with the latest consistent
-snapshot.
-
-All clients, such as pip, using the TUF protocol MUST be modified to download
-every metadata and target file (except for *timestamp* metadata) by including,
-in the file request, the version of the file (for metadata), or the
-cryptographic hash of the file (for target files) in the filename.
-
-Finally, PyPI SHOULD use a `transaction log`__ to record project transaction
-processes and queues so that it will be easier to recover from errors after a
-server failure.
+When a new project release is uploaded to PyPI, PyPI MUST update the *bin-n*
+metadata responsible for the target files of the project release (remember that
+target files are sorted into bins by their filename hashes). Consequentially,
+PyPI MUST update *snaphot* to account for the updated *bin-n* metadata, and
+*timestamp* to account for the updated *snapshot* metadata. These updates
+SHOULD be handled by automated processes, e.g. one or more *transaction
+processes* and one *snapshot process*.
```

Comment:
Can you define the transaction processes and explain how many new targets should be included in each process?